core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -86,20 +90,56 @@ public TableScan appendsAfter(long fromSnapshotId) {
   public TableScan useSnapshot(long scanSnapshotId) {
     Preconditions.checkArgument(
         snapshotId() == null, "Cannot override snapshot, already set to id=%s", snapshotId());
+    Preconditions.checkArgument(
+        ref() == null, "Cannot override snapshot, already set to id=%s", ref());
```

Comment:
Looks like this should update `id=` to `ref=`.

Reply:
I think we can actually remove this validation; if a ref is set we always set a snapshot as well. This snapshot can either be a tag, tip of a branch, or the snapshot from a time travel on the branch. 

The snapshot check on the line before is sufficient validation; although the error message won't indicate the ref is set. I can include that in the error message if that's desired?